include/sparrow/layout/mutable_array_base.hpp

```diff
@@ -242,79 +260,102 @@ namespace sparrow
-     */
     template <class D>
-    auto mutable_array_base<D>::insert(const_iterator pos, const value_type& value, size_type count) -> iterator
+    template <typename T>
+    auto mutable_array_base<D>::insert(const_iterator pos, const nullable<T>& value, size_type count)
+        -> iterator
     {
         SPARROW_ASSERT_TRUE(pos >= this->cbegin());
         SPARROW_ASSERT_TRUE(pos <= this->cend());
         const size_t distance = static_cast<size_t>(std::distance(this->cbegin(), pos));
         auto& derived = this->derived_cast();
         derived.insert_bitmap(sparrow::next(this->bitmap_cbegin(), distance), value.has_value(), count);
         derived.insert_value(sparrow::next(derived.value_cbegin(), distance), value.get(), count);
-        this->get_arrow_proxy().set_length(this->size() + count);  // Must be done after resizing the bitmap
-                                                                   // and values
+        this->get_arrow_proxy().set_length(this->size() + count);  // Must be done after resizing the
+                                                                   // bitmap and values
         derived.update();
         return sparrow::next(this->begin(), distance);
     }
 
-    /**
-     * Inserts elements from initializer list \c values before \c pos in the array.
-     *
-     * @param pos The iterator before which the elements will be inserted (\c pos may be the end() iterator).
-     * @param values The <tt>std::initializer_list</tt> to insert the values from.
-     * @return An iterator pointing to the first element inserted, or \c pos if \c values is empty.
-     */
     template <class D>
-    auto mutable_array_base<D>::insert(const_iterator pos, std::initializer_list<value_type> values) -> iterator
+    template <typename T>
+    auto mutable_array_base<D>::insert(const_iterator pos, std::initializer_list<nullable<T>> values)
+        -> iterator
     {
         return insert(pos, values.begin(), values.end());
     }
 
-    /**
-     * Removes the element at \c pos from the array.
-     *
-     * @param pos The iterator to the element to remove.
-     * @return The iterator following the last element removed.
-     */
+    template <class D>
+    template <typename InputIt>
+        requires std::input_iterator<InputIt>
+                 && mpl::is_type_instance_of_v<typename std::iterator_traits<InputIt>::value_type, nullable>
+    auto mutable_array_base<D>::insert(const_iterator pos, InputIt first, InputIt last) -> iterator
+    {
+        SPARROW_ASSERT_TRUE(pos >= this->cbegin())
+        SPARROW_ASSERT_TRUE(pos <= this->cend());
+        SPARROW_ASSERT_TRUE(first <= last);
+        const difference_type distance = std::distance(this->cbegin(), pos);
+        const auto validity_range = std::ranges::subrange(first, last)
+                                    | std::views::transform(
+                                        [](const auto& obj)
```

Comment:
This modification is made to accept nullable of type. Then we let the `derived.insert_values `accept or not the inner type of the nullable.
In my case I want to be able to accept any kind of range with the right inner type